Source/CBL_ForestDBStorage.mm

```diff
@@ -992,8 +998,11 @@ - (CBL_Revision*) addDocID: (NSString*)inDocID
         return (CBLStatus)status;
     }];
 
-    if (CBLStatusIsError(*outStatus))
+    if (CBLStatusIsError(*outStatus)) {
+        if (outError && !*outError)
```

Comment:
Oog, this is messy. I think you'll need to initialize `*outError` to nil at the top of the method.